modules/aws_base/tf_module/s3_gateway.tf

```diff
@@ -0,0 +1,15 @@
+resource "aws_vpc_endpoint" "s3" {
```

Comment:
Should we gate this on `local.create_vpc` as well? Does this serve any purpose without the accompanying `aws_vpc_endpoint_route_table_association`? If a user is specifying an external VPC ID, would it perhaps be reasonable to require that they create the S3 VPC endpoint on their end first?

Reply:
@jeevb I was just about to leave a similar comment thinking through this.

I agree with you.  If users are creating their own VPC, it should be up to them to manage the endpoints.  We should gate both the `aws_vpc_endpoint` and `aws_vpc_endpoint_route_table_association` resources.

Incoming fix...